sync/worker_pool.go

```diff
@@ -56,6 +56,9 @@ func NewWorkerPool(size int) WorkerPool {
 }
 
 func (p *workerPool) Init() {
+	if len(p.workCh) == cap(p.workCh) {
```

Comment:
Since the length of the worker channel can change as the pool is used would it be better to instead use an internal flag and only initialize the channel if the flag is not?

Reply:
The work channel is not public, so it would have to change within this package, and there is no code to do that right now. 

But in the future, I suppose it could change if some functionality was added. I can use a flag.

Reply:
I just meant that if one calls `Go` on the worker pool that will decrement the length of the channel until the worker is put back on the queue. So if there are any concurrent calls to `Go` between calls to `Init` you'll run into the same issue.